#goal definition
char[] outputMessage
char[] userOptions
---
#result definition
bool isOK
---
#feedback
int32[] sequence
